Ios scroll fade

simarjot0032 opened the pull request:
- Fixed the Internet checking method for better perfomance for ios
- Changed the approach from the setTimeout and setInterval to Animated.timing 
- Fixed the scroll and animation in IOS

simarjot0032 committed:
- fixed the fade animation bug on ios

simarjot0032 committed:
- converted the network check for the better clean up

simarjot0032 committed:
- better internet checking for better JS thread handling

simarjot0032 committed:
- fix: path saving properly in ios

simarjot0032 committed:
Merge branch 'ios-build' of github.com:KhalisFoundation/my-sehaj-path-app into ios-scroll-fade

simarjot0032 committed:
- fade animation and the auto scrolled fixed

Gauravjeetsingh commented on components/AngsNavigation.tsx:
```diff
@@ -52,43 +52,45 @@ export const AngsNavigation = ({
+      .then((isConnected) => {
+        if (!isConnected) {
+          showErrorAlert(
+            ErrorConstants.NO_INTERNET_TITLE + '\n' + ErrorConstants.NO_INTERNET_MESSAGE
```
We can use template string here.

```javascript
`${ErrorConstants.NO_INTERNET_TITLE}\n ${ErrorConstants.NO_INTERNET_MESSAGE}`
```

Gauravjeetsingh commented on components/SimpleTextForPath.tsx:
```diff
@@ -68,22 +60,21 @@ export const SimpleTextForPath = ({
+      onSelection();
+      setIsLongPressing(true);
+      setIsSaving(true);
+      setIsSaved(true);
```
What is difference between states setIsSaving and setIsSaved?

Is there any use case, where these two would have different value?

In this case, both are set to true at same time.

simarjot0032 replied:
Yes veerji we have 2 states for saving panktee
1- for when user click on save button which show msg 'Select a panktee to save'
2- when user select which shows msg 'Select panktee saved'
but when user tries to saves using the long press we dont show the 'Select a panktee to save'

Gauravjeetsingh replied:
Got it

Gauravjeetsingh commented on hooks/useInternet.ts:
```diff
@@ -4,27 +4,53 @@ import { useState, useEffect } from 'react';
+        })
+        .catch((error) => {
+          clearTimeout(timeout);
+          console.error('Error in checkNetwork:', error);
```
We can remove this console message

Gauravjeetsingh replied:
There are bunch of other console messages in the code, can you take care of those too, if they aren't added for any specific reason.

Gauravjeetsingh commented on hooks/useLocal.ts:
```diff
@@ -138,14 +138,18 @@ export const useLocal = () => {
-        await AsyncStorage.setItem('pathDateDetails', JSON.stringify(updatedPathDate));
         setIsSaved(true);
       } else {
+        console.error('Failed to find matched path or date');
```
Any reason on why we need the console messages here?

Gauravjeetsingh commented on hooks/useNavigation.ts:
```diff
@@ -30,31 +31,35 @@ export const useNavigation = ({
       try {
+        const isConnected = await checkNetwork();
+        if (!isConnected) {
+          console.log('showErrorAlert');
```
Another console message that can be removed.

Gauravjeetsingh commented on hooks/useScrollToSavedPath.ts:
```diff
@@ -31,36 +30,36 @@ export const useScrollToSavedPath = ({
+      }),
+    ]).start(() => {
+      setIsSaving(false);
+      setIsSaved(false);
```
Another place, where both setIsSaving and setIsSaved are assigned same value at same place.

simarjot0032 replied:
yes veer ji this is when the path is loaded it shows the Last saved panktee found for that other are being false

Gauravjeetsingh commented on screens/PathScreen.tsx:
```diff
@@ -47,26 +48,40 @@ export const PathScreen = ({ navigation, route }: PathScreenProps) => {
+  const scrolledToSavedPath = useRef<boolean>(false);
   const scorllOffset = useRef<number>(0);
   const scrollRef = useRef<ScrollView | null>(null);
   const aleartIndicator = useRef<React.ReactNode>();
```
Misspelled alert in above variable.

Gauravjeetsingh reviewed (changes requested):
Added few comments, have a look

simarjot0032 committed:
- fixed the typo mistake
- removed the console statements
- use the template strings insted of concatenation
